www/base/src/app/changes/changebuilds/changebuilds.controller.js

```diff
@@ -0,0 +1,38 @@
+class ChangeBuildsController {
+    constructor($scope, dataService, bbSettingsService, $stateParams, resultsService, $interval, restService) {
+
+        _.mixin($scope, resultsService);
+        $scope.settings = bbSettingsService.getSettingsGroup('ChangeBuilds');
+        $scope.$watch('settings', () => bbSettingsService.save()
+        , true);
+        const buildsFetchLimit = $scope.settings.buildsFetchLimit.value;
+
+        const dataAccessor = dataService.open().closeOnDestroy($scope);
+        $scope.builders = dataAccessor.getBuilders();
+
+        const changeId = $scope.changeId = $stateParams.changeid;
+
+        const stop = $interval(() => {
+            let requestUrl = `changes/${changeId}/builds`;
+            if (!buildsFetchLimit == '') {
+                requestUrl = `changes/${changeId}/builds?limit=${buildsFetchLimit}`;
+            }
+            restService.get(requestUrl).then((data) => {
+                $scope.builds = data.builds;
+            });
+        }, 5000);
```

Comment:
I think with this, you only get first update after 5s.
Lets factorize the update function, and call it once at init.